test/unit/collector/api-connect.test.js

```diff
@@ -633,6 +628,94 @@ tap.test('retries on receiving invalid license key (401)', (t) => {
   }
 })
 
+tap('retries on misconfigured proxy', (t) => {
+  const sandbox = sinon.createSandbox()
+  const loggerMock = require('../mocks/logger')(sandbox)
+  const CollectorApiTest = proxyquire('../../../lib/collector/api', {
+    '../logger': {
+      child: sandbox.stub().callsFake(() => loggerMock)
+    }
+  })
+  t.autoend()
+
+  let collectorApi = null
+  let agent = null
+
+  const error = {
+    code: 'EPROTO'
+  }
+
+  let failure = null
+  let success = null
+  let connect = null
+
+  t.beforeEach((done) => {
+    fastSetTimeoutIncrementRef()
+
+    nock.disableNetConnect()
+
+    agent = setupMockedAgent()
+    agent.config.proxy_port = '8080'
+    agent.config.proxy_host = 'test-proxy-server'
+    collectorApi = new CollectorApiTest(agent)
+
+    const preconnectURL = helper.generateCollectorPath('preconnect')
+    failure = nock(URL).post(preconnectURL).times(1).replyWithError(error)
+    success = nock(URL).post(preconnectURL).reply(200, {return_value: {}})
+    connect = nock(URL)
+      .post(helper.generateCollectorPath('connect'))
+      .reply(200, {return_value: {agent_run_id: 31338}})
+
+    done()
+  })
+
+  t.afterEach((done) => {
+    sandbox.resetHistory()
+    restoreSetTimeout()
+
+    if (!nock.isDone()) {
+      /* eslint-disable no-console */
+      console.error('Cleaning pending mocks: %j', nock.pendingMocks())
+      /* eslint-enable no-console */
+      nock.cleanAll()
+    }
+
+    nock.enableNetConnect()
+    helper.unloadAgent(agent)
+
+    done()
+  })
+
+  t('should log warning when proxy is misconfigured', (t) => {
+    testConnect(t, () => {
+      t.deepEqual(loggerMock.warn.args, [
+        [
+          error,
+          'Your proxy server appears to be configured to accept connections over http. When setting `proxy_host` and `proxy_port` New Relic attempts to connect over SSL(https). If your proxy is configured to accept connections over http, try setting `proxy` to a fully qualified URL(e.g http://proxy-host:8080).'
```

Comment:
surprised linter didn't catch this. Possible to wrap text? or just put in `expectedErrorMsg` var?

Reply:
I think it's because we have a diff .eslintrc for tests